### ISSUE TITLE
Can not build cabal with stack

### ISSUE THREAD

**Issue by flip111** [haskell/cabal#6525]
```
$ git clone https://github.com/haskell/cabal.git
$ cd cabal
$ stack build --ghc-options="-Wall"

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for cabal-install-3.3.0.0:
    lukko must match >=0.1 && <0.2, but the stack configuration has no specified version  (latest matching version is 0.1.1.1)
    windns must match >=0.1.0 && <0.2, but the stack configuration has no specified version  (latest matching version is 0.1.0.1)
needed since cabal-install is a build target.

In the dependencies for cabal-testsuite-3:
    Cabal-3.3.0.0 from stack configuration does not match ((==2.2.* || ==2.4.* || ==3.0.*) && ==3.3.0.0) && (==2.2.* || ==2.4.* || ==3.0.*)
    aeson-1.2.4.0 from stack configuration does not match ^>=1.4.2.0  (latest matching version is 1.4.6.0)
    async-2.1.1.1 from stack configuration does not match >=2.2.1 && <2.3  (latest matching version is 2.2.2)
    exceptions-0.8.3 from stack configuration does not match >=0.10.0 && <0.11  (latest matching version is 0.10.4)
    optparse-applicative-0.14.2.0 from stack configuration does not match >=0.14.3.0 && <0.15 || >=0.15.1.0 && <0.16  (latest matching version is 0.15.1.0)
    regex-tdfa-1.2.3 from stack configuration does not match ^>=1.2.3.1 || ^>=1.3.1.0  (latest matching version is 1.3.1.0)
    temporary-1.2.1.1 from stack configuration does not match ^>=1.3  (latest matching version is 1.3)
    text-1.2.3.0 from stack configuration does not match ^>=1.2.3.1  (latest matching version is 1.2.4.0)
needed since cabal-testsuite is a build target.

In the dependencies for hackage-security-0.5.3.0:
    Cabal-3.3.0.0 from stack configuration does not match >=1.14 && <2.4  (latest matching version is 2.2.0.1)
needed due to cabal-install-3.3.0.0 -> hackage-security-0.5.3.0

Some different approaches to resolving this:

  * Recommended action: try adding the following to your extra-deps in C:\Users\philippe\haskell\cabal\stack.yaml:

- Cabal-2.2.0.1@sha256:2a80d8fb655474f0eaeb20434c47f64f84e6302e55973056f00df8ca050b9683,20037
- aeson-1.4.6.0@sha256:560575b008a23960403a128331f0e59594786b5cd19a35be0cd74b9a7257958e,6980
- async-2.2.2@sha256:ed46f0f5be36cf8a3e3aebc6827d015e1f3bf9615c245e057b9e9bd35faddd21,2895
- exceptions-0.10.4@sha256:d3aaa8d112cd53e16b5991bf5e43237ad6cf8277b8bab5cb8ab8e081dbb556f7,2684
- lukko-0.1.1.1@sha256:5c674bdd8a06b926ba55d872abe254155ed49a58df202b4d842b643e5ed6bcc9,4289
- optparse-applicative-0.15.1.0@sha256:a0b9924d88a17c36cd8e6839d7dd2138419dd8f08cbb4f9af18f3c367b0c69a3,4673
- regex-tdfa-1.3.1.0@sha256:c77808a0d68d275c75fb84dc9ced340536574c9aa7961604036a606a232d585b,6274
- temporary-1.3@sha256:3a66c136f700dbf42f3c5000ca93e80b26dead51e54322c83272b236c1ec8ef1,1485
- text-1.2.4.0@sha256:8c24450feb8e3bbb7ea3e17af24ef57e85db077c4bf53e5bcc345b283d1b1d5b,10081
- windns-0.1.0.1@sha256:4f881c89919bbdae91e028244e7dd0c2fca572e3b9d5f87cdebe6012ff25a49d,2127

Plan construction failed.
```

**Comment by phadej** [comment]
`stack.yaml` is not actively maintained. PRs to fix it are welcome.

**Comment by phadej** [comment]
Fixed in #6527, I hope.
